### PR TITLE
Display order type (limit|market|liquidity)

### LINKED ISSUE #1346: [LIMIT] [7-Explorer] Show Order Type
Show the type of order.

There's 3 types of orders

## Tasks
- [ ] Identify and model the type of orders (get it from backend)
- [ ] Represent the type of order in the Explorer

## Represent the type of order
There could be multiple ways to represent the type.

@alfetopito suggested:  
```
The first idea that I propose is to replace the text with labels for all properties, where there would be 3 types of tags in the same type:

1. SWAP | LIMIT | LIQUIDITY
2. SELL | BUY
3. FILL OR KILL | PARTIAL FILL

Then the `type` field will be a combination of all 3.
```

@fairlighteth  what do you think

## Details
The backend will now include a order type in the order details (`GET /orders/{orderId}`)

```js
{
	orderType: Enum<MARKET | LIMIT | LIQUIDITY>
}
```

We will show the new information in:

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Closes https://github.com/cowprotocol/cowswap/issues/1346

There are three types of orders: `limit|market|liquidity`.
The label of type has format: `{Sell|buy} {limit|market|liquidity} order <(Fill or Kill)>`

image

image

image

image

image

image

### CODE REVIEW

**Comment by elena-zh** [comment]
Order IDs for tests: 
 - Sell market order: https://pr259--explorer.review.gnosisdev.com/goerli/orders/0xc5baa7f0f44e5693968aaf79cb9d987972a121cacff8c6f244c99e27c50a4f542db8323b3766b868aaf6a967e55eeb35c233de9d63765271
 - Buy market order: https://pr259--explorer.review.gnosisdev.com/goerli/orders/0xb89875370e76fb9bc71e7065fa0e14ed5e77782c28f91c0e77c1e0a1ac4995f32db8323b3766b868aaf6a967e55eeb35c233de9d63764c17
 - Buy limit order: https://pr259--explorer.review.gnosisdev.com/goerli/orders/0xf7495ce1534883f9d9ec6afd891b23bffed30621af76c72118efb68ae1fadff82db8323b3766b868aaf6a967e55eeb35c233de9d6376583e
 - Sell limit order: https://pr259--explorer.review.gnosisdev.com/goerli/orders/0x3d54fa9d46ddbb74b80eba063b8ef9e50ae2aff85aa9d057ff581d522d9466322db8323b3766b868aaf6a967e55eeb35c233de9d63764b8e/
 - Sell Liquidity order: https://pr259--explorer.review.gnosisdev.com/orders/0x4a3927813f512fcc6ba710ee44d2a9204d21e997c5c122a60d1d4a2106fdd848042b32ac6b453485e357938bdc38e0340d4b927661c2639b/
 - Buy Liquidity order: https://pr259--explorer.review.gnosisdev.com/orders/0x93299f3754e51938baae1946583382f9906092213cf8042b81eb877c8923fb67e63a13eedd01b624958acfe32145298788a7a7ba624ae706

**Comment by alfetopito** [comment]
I'm pretty sure I had a comment on the specs about it here https://docs.google.com/document/d/1P9I4IkfsNXjyhzahhLGequaNKoGik0jP9lTO4TIBNfw/edit#heading=h.tzhsrhom2byf

But it's not there anymore...

Anyway, my proposal was to replace the full text version with tags for each of the 3 properties:

1. MARKET (or SWAP), LIMIT and LIQUIDITY
2. SELL or BUY
3. FILL OR KILL or PARTIALLY FILLABLE

So we'd have and order with the tags: [MARKET][SELL][FILL OR KILL] and so on
That would apply also to list views, although can be treated in a different task/PR

@anxolin and @fairlighteth agreed with the suggestion (although I can't find the thread where this was discussed)

File: src/components/orders/DetailsTable/index.tsx [hunk 1]
```diff
@@ -82,7 +82,19 @@ const tooltip = {
...
+      <ul>
+        <li>A market order is an order to buy or sell at the market&apos;s current best available price</li>
+        <li>A limit order is an order to buy or sell at an arbitrary price specified by the user</li>
+        <li>A liquidity order is an order that market makers can provide as a source of liquidity</li>
+      </ul>
+      In addition, an order can be &quot;fill or kill&quot; or &quot;partially fillable&quot;. Currently all order types
+      can be &quot;fill or kill&quot; – only market orders can be either &quot;fill or kill&quot; or &quot;partially
+      fillable&quot;.
+    </div>
+  ),
```
Comment (anxolin) [inline src/components/orders/DetailsTable/index.tsx:97]: nit: For every list item, i would add the name of the order in bold, and then explain.

* **Market Order**:  Buy or Sell at ...
* **Limit Order**:  Buy or Sell at ...
* **Liquidity Order**:  Source of liquidity coming from market makers...

Also: 

image


I think this last part is a bit confusing. 

Orders can allow partial execution or not, there's two types:
- **Fill Or Kill**: Either the order is fully filled, or not filled at all. Currently all Market and Limit Orders are fill or kill. Support for partials will come soon!
- **Partial Execution**: The order can be executed partially as long as the limit price is respected.


Anyways, maybe @MindyCoW @avsavsavs have a better input in the specific message. 

As Leandro mentioned, we will need to break this type into 3 different concepts/tags (1. buy/sell, 2. partial/fill-kill, 3. market/limit/liquidity), each could have a tooltip to further clarify.

**Review by anxolin — APPROVED** [review]
Approving, but we should re-iterate (in another PR?) to do what we mentioned about splitting into 3 types (Leandro's comment above)

File: src/components/orders/DetailsTable/index.tsx (hunk 1 above)
Comment (avsavsavs) [reply src/components/orders/DetailsTable/index.tsx:97]: I think this is what made more sense to me: 
`In addition, an order can be "Fill, Kill or Partially Fillable". Currently only market orders are allowed to be partially fillable, while the other orders are just fill or kill.`

But Alex M suggested the copy you have in here[https://cowservices.slack.com/archives/C0361CDG8GP/p1669220826163489?thread_ts=1669206785.099929&cid=C0361CDG8GP](url)

**Comment by elena-zh** [comment]
> Approving, but we should re-iterate (in another PR?) to do what we mentioned about splitting into 3 types (Leandro's comment above)

I have created #262 issue for this

File: src/components/orders/DetailsTable/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/orders/DetailsTable/index.tsx:97]: there was a problem witht he link above. Adding it again https://cowservices.slack.com/archives/C0361CDG8GP/p1669220826163489?thread_ts=1669206785.099929&cid=C0361CDG8GP


@avsavsavs this is not true `Currently only market orders are allowed to be partially fillable` (it's liquidity orders), but I see is wrong in the original, and AlexM description too

Comment (anxolin) [reply src/components/orders/DetailsTable/index.tsx:97]: Let's settle this topic in the thread of Slack.

**Pull request merged**